### PR TITLE
Add vectorstore agent

### PR DESCRIPTION
**Pull request by agola11**
VectorStoreQAWithSourcesChain not supported in ts yet, so only added VectorStoreQA

### CODE REVIEW

File: langchain/src/agents/tools/vectorstore.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  static getDescription(name: string, description: string): string {
+    return `Useful for when you need to answer questions about ${name}. "
+            "Whenever you need information about ${description} "
+            "you should ALWAYS use this. "
+            "Input should be a fully formed question.`;
+  }
+
+  async call(input: string) {
+    const chain = VectorDBQAChain.fromLLM(this.llm, this.vectorStore);
```
Comment (nfcampos) [inline langchain/src/agents/tools/vectorstore.ts:36]: Is it on purpose that we create a new chain for each call?

File: langchain/src/agents/tools/vectorstore.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  constructor(name: string, description: string, fields: VectorStoreTool) {
+    super();
+    this.name = name;
+    this.description = description;
+    this.vectorStore = fields.vectorStore;
+    this.llm = fields.llm;
+  }
+
+  static getDescription(name: string, description: string): string {
+    return `Useful for when you need to answer questions about ${name}. "
```
Comment (nfcampos) [inline langchain/src/agents/tools/vectorstore.ts:29]: I think the `"` shouldn't be here? In a ` string double quotes will just show up as any other character (so will the whitespace at the beginning of each line).

**Review by nfcampos — COMMENTED** [review]
This looks great!

File: langchain/src/agents/tools/vectorstore.ts (hunk 1 above)
Comment (agola11) [reply langchain/src/agents/tools/vectorstore.ts:36]: yeah good point, can just initialize this in constructor, was just blindly doing what we do in python

File: langchain/src/agents/tools/vectorstore.ts (hunk 2 above)
Comment (agola11) [reply langchain/src/agents/tools/vectorstore.ts:29]: yup good catch

**Pull request merged**